pointcloud/pointcloud_file_test.go

```diff
@@ -321,6 +329,17 @@ func TestPCDKDTree(t *testing.T) {
 	test.That(t, gotPt, test.ShouldNotBeNil)
 }
 
+func TestPCDOctree(t *testing.T) {
+	gotPCD := createNewPCD(t)
+
+	basicOct, err := ReadPCDToBasicOctree(strings.NewReader(gotPCD))
```

Comment:
Is there any way to assert that the returned value is an octree & that it implements the pointcloud interface? If so, that seems like a good test to prove that the acceptance criteria of this ticket was met.

Reply:
That is an inherent check of the NewBasicOctree function as it returns a PointCloud, if the interface is not implement not only would tests fail but go would fail to compile it

Reply:
I'm recommending changing all the functions which return a `Pointcloud` to return `BasicOctree` instead (b/c returning a different type of pointcloud would differ from the end user's expectations. 

After that refactor, could this test (or maybe a different test) verify that BasicOctree implements the Pointcloud interface?